fixieai/cli/agent/commands.py

```diff
@@ -138,71 +274,51 @@ def _update_agent_requirements(
     type=str,
     help="Additional requirements for requirements.txt. Can be specified multiple times.",
 )
-def init_agent(handle, description, entry_point, more_info_url, requirement):
+def init_agent(handle, description, entry_point, more_info_url, requirement, language):
+    templator: AgentTemplator
+    if language.lower() in ["python", "py"]:
+        templator = PythonTemplator()
+    elif language.lower() in ["typescript", "ts"]:
+        templator = TypeScriptTemplator()
+
     try:
         current_config = agent_config.load_config()
     except FileNotFoundError:
-        current_config = agent_config.AgentConfig()
+        current_config = agent_config.AgentConfig(language=language)
     current_config.handle = handle
     current_config.description = description
     current_config.entry_point = entry_point
+    current_config.language = language
     current_config.more_info_url = more_info_url
+
+    # Having this hook here is maybe a little gross.
+    templator.modify_new_agent_config(current_config)
+
     agent_config.save_config(current_config)
 
     entry_module, _ = entry_point.split(":")
-    expected_main_path = entry_module.replace(".", "/") + ".py"
+    expected_main_path = (
+        entry_module.replace(".", "/") + templator.get_main_file_extension()
+    )
     if not os.path.exists(expected_main_path):
-        urllib.request.urlretrieve(AGENT_TEMPLATE_URL, expected_main_path)
+        urllib.request.urlretrieve(
+            templator.get_agent_template_url(), expected_main_path
+        )
         click.secho(
             f"Initialized agent.yaml and made a template agent file at {expected_main_path}",
             fg="green",
         )
     else:
         click.secho(f"Initialized agent.yaml.", fg="green")
 
-    try:
-        with open(REQUIREMENTS_TXT, "rt") as requirements_txt:
-            existing_requirements = list(
-                r.strip() for r in requirements_txt.readlines()
-            )
-    except FileNotFoundError:
-        existing_requirements = []
-
-    resolved_requirements = _update_agent_requirements(
-        existing_requirements, list(requirement)
+    templator.write_helper_files(
```

Comment:
IMO it's probably simpler to structure this as:

```python
def init_agent(...):
  # setup the config
  agent_config.foo = bar #etc.
  
  if language == "python":
     init_python_agent(config)
  elif language == "typescript":
     init_typescript_agent(config)
```

Even though there'll probably some similar logging code or whatever between the two functions I don't think they share enough to make the templator abstraction worth it. WDYT?